src/scanner.cc

```diff
@@ -172,7 +172,9 @@ static vector<string> names = {
   "empty",
 };
 
-static string name(Sym t) { return t < names.size() ? names[t] : "unknown"; }
+static const int names_size = sizeof(names) / sizeof(names[0]);
```

Comment:
Is this correct? Maybe just write `names_size = ...` with ... being the actual count. If you make it constexpr you can use it in `std::array` too.

Reply:
It's correct. This is a standard way of finding the size of a constant array in c.